packages/swapper/src/swappers/zrx/approvalNeeded/approvalNeeded.test.ts

```diff
@@ -0,0 +1,115 @@
+import Web3 from 'web3'
+import { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import { ChainTypes } from '@shapeshiftoss/types'
+import { ChainAdapterManager } from '@shapeshiftoss/chain-adapters'
+import { approvalNeeded } from './approvalNeeded'
+import { setupQuote } from '../utils/test-data/setupSwapQuote'
+import { zrxService } from '../utils/zrxService'
+import { APPROVAL_GAS_LIMIT } from '../utils/constants'
+
+jest.mock('web3')
+jest.mock('axios', () => ({
+  create: jest.fn(() => ({
+    get: jest.fn()
+  }))
+}))
+
+// @ts-ignore
+Web3.mockImplementation(() => ({
+  eth: {
+    Contract: jest.fn(() => ({
+      methods: {
+        allowance: jest.fn(() => ({
+          call: jest.fn()
+        }))
+      }
+    }))
+  }
+}))
+
+const setup = () => {
+  const unchainedUrls = {
+    [ChainTypes.Ethereum]: 'http://localhost:31300/api/v1'
+  }
+  const ethNodeUrl = 'http://localhost:1000'
+  const adapterManager = new ChainAdapterManager(unchainedUrls)
+  const web3Provider = new Web3.providers.HttpProvider(ethNodeUrl)
+  const web3 = new Web3(web3Provider)
+
+  return { web3, adapterManager }
+}
+
+describe('approvalNeeded', () => {
+  const { web3, adapterManager } = setup()
+  const args = { web3, adapterManager }
+  const walletAddress = '0xc770eefad204b5180df6a14ee197d99d808ee52d'
+  const wallet = ({
+    ethGetAddress: jest.fn(() => Promise.resolve(walletAddress))
+  } as unknown) as HDWallet
+
+  const { quoteInput, sellAsset } = setupQuote()
+
+  it('returns false if sellAsset symbol is ETH', async () => {
+    const input = {
+      quote: { ...quoteInput, sellAsset: { ...sellAsset, symbol: 'ETH' } },
+      wallet
+    }
+
+    expect(await approvalNeeded(args, input)).toEqual({ approvalNeeded: false })
+  })
+
+  it('returns false if sellAsset chain is not ETH', async () => {
```

Comment:
not sure if this test is valid if 0x swapper only deals with eth?

Reply:
True, but once we implement other chain adapters there will be a chance where a different chain type can be passed in the `sellAsset`. This is just testing the guard clause so we don't send a receive address from the wrong chain.

Reply:
ok, ya so let's throw rather than return false if we pass a non eth chain asset to this swapper